x-pack/plugins/stack_alerts/public/alert_types/es_query/search_source_threshold_expression.tsx

```diff
@@ -0,0 +1,208 @@
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  EuiSpacer,
+  EuiTitle,
+  EuiExpression,
+  EuiPopover,
+  EuiText,
+  EuiLoadingSpinner,
+  EuiEmptyPrompt,
+} from '@elastic/eui';
+import { Filter, ISearchSource } from '../../../../../../src/plugins/data/common';
+import { QueryStringInput } from '../../../../../../src/plugins/data/public';
+import { EsQueryAlertParams } from './types';
+import {
+  ForLastExpression,
+  RuleTypeParamsExpressionProps,
+  ThresholdExpression,
+} from '../../../../triggers_actions_ui/public';
+import { DEFAULT_VALUES } from './constants';
+import { ReadOnlyFilterItems } from './read_only_filter_items';
+
+export const SearchSourceThresholdExpression = ({
+  ruleParams,
+  setRuleParams,
+  setRuleProperty,
+  data,
+  errors,
+}: RuleTypeParamsExpressionProps<EsQueryAlertParams>) => {
+  const { thresholdComparator, threshold, timeWindowSize, timeWindowUnit } = ruleParams;
+
+  const getDefaultParams = () => {
+    const defaults = {
+      timeWindowSize: timeWindowSize ?? DEFAULT_VALUES.TIME_WINDOW_SIZE,
+      timeWindowUnit: timeWindowUnit ?? DEFAULT_VALUES.TIME_WINDOW_UNIT,
+      threshold: threshold ?? DEFAULT_VALUES.THRESHOLD,
+      thresholdComparator: thresholdComparator ?? DEFAULT_VALUES.THRESHOLD_COMPARATOR,
+    };
+
+    return { ...ruleParams, ...defaults, searchType: 'searchSource' };
+  };
+
+  const [currentAlertParams, setCurrentAlertParams] = useState<EsQueryAlertParams>(
+    getDefaultParams()
+  );
+
+  const setParam = useCallback(
+    (paramField: string, paramValue: unknown) => {
+      setCurrentAlertParams((currentParams) => ({
+        ...currentParams,
+        [paramField]: paramValue,
+      }));
+      setRuleParams(paramField, paramValue);
+    },
+    [setRuleParams]
+  );
+
+  useEffect(() => {
+    setRuleProperty('params', currentAlertParams);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const [usedSearchSource, setUsedSearchSource] = useState<ISearchSource | undefined>();
+
+  const { searchConfiguration } = ruleParams;
+
+  // Note that this PR contains a limited way to edit query and filter
+  // But it's out of scope for the MVP
+  const [showQueryBar, setShowQueryBar] = useState<boolean>(false);
+  const [showFilter, setShowFilter] = useState<boolean>(false);
+
+  useEffect(() => {
+    async function initSearchSource() {
+      const loadedSearchSource = await data.search.searchSource.create(searchConfiguration);
+      setUsedSearchSource(loadedSearchSource);
+    }
+    if (searchConfiguration) {
+      initSearchSource();
+    }
+  }, [data.search.searchSource, searchConfiguration]);
+
+  if (!usedSearchSource) {
+    return (
+      <EuiEmptyPrompt
+        title={<EuiLoadingSpinner size="xl" />}
+        body={
+          <EuiText color="subdued">
+            <FormattedMessage
+              id="xpack.stackAlerts.searchThreshold.ui.loadingPrompt"
+              defaultMessage="Loading rule type params…"
```

Comment:
There is currently a problem with two loadings for a moment. Would it be better to return `null` when `usedSearchSource` is not defined?

![C8257DF1-ECD9-4164-95D7-4F3601A89B2F](https://user-images.githubusercontent.com/39378793/153228430-d56093db-0100-4e2f-9602-048ba63c110e.jpeg)

Reply:
yes, one loading should be sufficient 👍

Reply:
I have found that two loadings appear before our expression rendered, so it should be `alert_form` bug. It is also happens when you open other rule expressions at first time, like `.es-query`.

Reply:
For our expression I would leave only loading without label. It will be displayed after two spinners will go away.